### PR TITLE
[ckeditor__ckeditor5-list] Add DocumentList and DocumentListProperties type definition

### PR DESCRIPTION
**Pull request by martin-badin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test ckeditor__ckeditor5-list`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/ckeditor/ckeditor5/tree/master/packages/ckeditor5-list>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by martin-badin** [comment]
Reping @fedemp

File: types/ckeditor__ckeditor5-list/src/documentlistproperties/documentlistpropertiesutils.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * Gets all the style types supported by given list type.
+     */
+    getAllSupportedStyleTypes(): string[];
+
+    /**
+     * Checks whether the given list-style-type is supported by numbered or bulleted list.
+     *
+     * @returns {'bulleted'|'numbered'|null}
```
Comment (fedemp) [inline types/ckeditor__ckeditor5-list/src/documentlistproperties/documentlistpropertiesutils.d.ts:17]: No need for JSDoc here.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ckeditor__ckeditor5-list ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ckeditor__ckeditor5-list@latest/))
was missing the following properties:
1. DocumentListPropertiesEditing
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c156e8f7d9088795427670b2b1a792e22df1693e
</p>

**Review by fedemp — APPROVED** [review]

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**